src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -492,8 +500,9 @@ private static void validate(SelectStatement prepared)
         if (prepared.hasAggregation())
             throw invalidRequest(NO_AGGREGATION_IN_TXNS_MESSAGE, "SELECT", prepared.source);
 
+        // when "LIMIT ?" this check can't be performed, so need to do again once the options are known
         if (prepared.getRestrictions().keyIsInRelation())
-            checkTrue(prepared.getLimit(null) == DataLimits.NO_LIMIT, NO_PARTITION_IN_CLAUSE_WITH_LIMIT, "SELECT", prepared.source);
+            checkTrue(prepared.isLimitMarker() || prepared.getLimit(null) == DataLimits.NO_LIMIT, NO_PARTITION_IN_CLAUSE_WITH_LIMIT, "SELECT", prepared.source);
```

Comment:
when `LIMIT ?` this would cause a NPE as `options` is `null`... we only don't NPE when the value is a literal...